HashRegistrar.sol

```diff
@@ -0,0 +1,320 @@
+        hash.status = Mode.Auction;        
+    }
+
+    // Allows you to open multiple for better anonimity
+    function startAuctions(bytes32[] _hashes) noEther{
+        for (uint i = 0; i < _hashes.length; i ++ ) {
+            startAuction(_hashes[i]);
+        }
+    }
+    
+    function shaBid(bytes32 hash, address owner, uint value, bytes32 salt) constant returns (bytes32 sealedBid) {
+        return sha3(hash, owner, value, salt);
+    }
+    
+    /*
+    ## One week auction for the desired hash
+
+    Bids are sent by sending a message to the main contract with a hash and an amount. The hash 
+    contains information about the bid, including the bidded hash, the bid amount, and a random 
+    salt. Bids are not tied to any one auction until they are revealed. The value of the bid 
+    itself can be maskeraded by changing the required period or sending more than what you are 
+    bidding for. This is followed by a 24h reveal period. Bids revealed after this period will 
+    be burned and the ether unrecoverable. Since this is an auction, it is expected that most 
+    public hashes, like known domains and common dictionary words, will have multiple bidders pushing the price up. 
+    */ 
+    function newBid(bytes32 sealedBid) {
+        if (address(sealedBids[sealedBid]) > 0 ) throw;
+        // creates a new hash contract with the owner
+        address newBid = new Deed(sealedBid, this);
+        sealedBids[sealedBid] = Deed(newBid);
+        if (!newBid.send(msg.value)) throw;
+    } 
+    
+    
+    /*
+    ## Winning bids are deposited
+
+    The highest bid gets control of the hash, by depositing the equivalent amount of the second 
+    highest bid. The funds will be held on new contract, controlled only by the owner which 
+    contains basic information like when the hash was first registered, when it needs to be 
+    renewed, and the price paid for it. **The price paid is also saved on a moving average amount, 
+    calculated as ```averagePrice = averagePrice * 0.999 + newPrice * 0.001```**. The 
+    averagePrice at the moment of purchase is also registered on the contract.
+    */ 
+    function revealBid(bytes32 _hash, address _owner, uint _value, bytes32 _salt) noEther  {
+        bytes32 seal = shaBid(_hash, _owner, _value, _salt);
+        Deed bid = sealedBids[seal];
+        if (address(bid) == 0 ) throw;
+        bid.unsealBid(_hash, _owner, _value, _salt);
+        sealedBids[seal] = Deed(0);
+        
+        entry hash = hashes[bid.hash()];
+        
+        if (bid.creationDate() > hash.auctionExpiration - revealPeriod
+            || now > hash.auctionExpiration ) {
+            // bid is invalid, refund 99.9%
+            bid.closeDeed(1);
+            
+        } else if ( _value < averagePrice / 100 ) {
+            // bid is invalid but not punishable
+            bid.closeDeed(1000);
+            
+        } else if (_value > hash.highestBid) {
+            // new winner
```

Comment:
This section needs to remove the deed from `sealedBids`, or the winner can wait until the auction is over, then 'reveal' their bid again and get a refund in the first clause of this if statement.